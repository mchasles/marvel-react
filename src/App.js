import React, { Component } from 'react';
import { HashRouter as Router, Route} from 'react-router-dom';
import {MarvelCharactersList} from './components/MarvelCharactersList/'
import {MarvelCharacterDetail} from './components/MarvelCharacterDetail/'
import {loadCharacters} from './lib/marvelApiService'
import './app.scss';

class App extends Component {
  state = {
    characters: []
  }

  componentDidMount() {
    loadCharacters()
      .then(characters => this.setState({characters}))
  }

  render() {
    const displayCharacters = this.state.characters;
    return (
      <Router>
        <div className="app">
          <MarvelCharactersList characters={displayCharacters}></MarvelCharactersList>
          <Route path='/character/:id' component={MarvelCharacterDetail} />
        </div>
      </Router>
    );
  }
}

export default App;
