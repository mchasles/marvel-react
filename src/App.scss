body {
  font-family: Arial, Helvetica, sans-serif;
  margin: 0;
}

a {
  cursor: pointer; 
}
.app {
  display: flex;
  background-color: #3b3b3b;
}

.app-landing-wrapper {
  flex-grow: 1;
  padding: 20px;
  position: relative;
}

.react-logo-landing {
  position: absolute;
  opacity: 0.05;
  top: 50%;
  transform: translate(-50%, -50%);
  left: 50%;
  height: 400px;
  width: 400px;
}
