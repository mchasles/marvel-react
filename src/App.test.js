import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

import renderer from 'react-test-renderer';

jest.mock('./lib/marvelApiService');

it('renders App without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('renders App correctly', () => {
  const app = renderer.create(
    <App />
  ).toJSON();
  expect(app).toMatchSnapshot();
});