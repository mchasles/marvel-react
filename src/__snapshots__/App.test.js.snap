exports[`test renders App correctly 1`] = `
<div
  className="app">
  <ul
    className="marvel-characters-list" />
</div>
`;
